src/NUnitFramework/framework/Constraints/Comparers/ComparisonState.cs

```diff
@@ -0,0 +1,36 @@
+using System.Collections.Generic;
+
+namespace NUnit.Framework.Constraints.Comparers
+{
+    internal struct ComparisonState
+    {
+        public struct Comparison
+        {
+            public object X { get; set; }
+            public object Y { get; set; }
+        }
+
+        /// <summary>
+        /// Flag indicating whether or not this is the top level comparison.
+        /// </summary>
+        public bool TopLevelComparison { get; set; }
+        public List<Comparison> Comparisons { get; set; }
```

Comment:
Please make this a private readonly field to make it easier to think about the possible mutations and to streamline the usability of this struct for future users.